cmd/snap-bootstrap/initramfs_systemd_mount.go

```diff
@@ -45,6 +46,90 @@ Wants=%[1]s
 	doSystemdMount = doSystemdMountImpl
 )
 
+// forbiddenChars is a list of characters that are not allowed in any mount paths used in systemd-mount.
+const forbiddenChars = `\,:" `
+
+// overlayFsOptions groups the options to systemd-mount related to overlayfs.
+type overlayFsOptions struct {
+	// Directories to be used as lower layers of an overlay mount.
+	// It does not need to be on a writable filesystem.
+	LowerDirs []string
+	// A directory to be used as the upper layer of an overlay mount.
+	// This is normally on a writable filesystem.
+	UpperDir string
+	// A directory to be used as the workdir of an overlay mount.
+	// This needs to be an empty directory on the same filesystem as upperdir.
+	WorkDir string
+}
+
+// Validate is used to perform consistency checks on the options related to overlayfs mounts
+func (o *overlayFsOptions) Validate() error {
+	if len(o.LowerDirs) <= 0 || len(o.UpperDir) <= 0 || len(o.WorkDir) <= 0 {
```

Comment:
Nitpick, for usability I'd split this into three checks.

You may also consider empty upperdir to be valid. When upperdir is empty, workdir is no longer needed and may be equally empty. See https://docs.kernel.org/filesystems/overlayfs.html#multiple-lower-layers

Reply:
Same as the previous comment. I split the checks and then tweaked them to allow for empty uppedir/workdir but throw an error if one is specified without the other.

Reply:
Nice